test/conformance/conformance_continue_node.cpp

```diff
@@ -18,236 +18,247 @@
+}
 
-    CopyCounterBody& operator=(const CopyCounterBody<O>& other){
-        copy_count = other.copy_count + 1;
-        return *this;
-    }
+//! Test body copying and copy_body logic
+//! Test the body object passed to a node is copied
+//! \brief \ref interface
+TEST_CASE("continue_node and body copying"){
+    conformance::test_copy_body<oneapi::tbb::flow::continue_node<int>, conformance::counting_object<int, oneapi::tbb::flow::continue_msg>>();
+}
 
-    O operator()(oneapi::tbb::flow::continue_msg){
-        return 1;
-    }
-};
+//! Test deduction guides
+//! \brief \ref interface \ref requirement
+TEST_CASE("Deduction guides"){
+#if __TBB_CPP17_DEDUCTION_GUIDES_PRESENT
+    test_deduction_guides();
+#endif
+}
 
-void test_copies(){
-    using namespace oneapi::tbb::flow;
+//! Test node broadcast messages to successors
+//! \brief \ref requirement
+TEST_CASE("continue_node broadcast"){
+    conformance::counting_functor<output_msg> fun(conformance::expected);
+    conformance::test_forwarding<oneapi::tbb::flow::continue_node<output_msg>,  oneapi::tbb::flow::continue_msg, output_msg>(1, fun);
+}
 
-    CopyCounterBody<int> b;
+//! Test node not buffered unsuccessful message, and try_get after rejection should not succeed.
+//! \brief \ref requirement
+TEST_CASE("continue_node buffering"){
+    conformance::dummy_functor<int> fun;
+    conformance::test_buffering<oneapi::tbb::flow::continue_node<int>, oneapi::tbb::flow::continue_msg>(fun);
+}
 
+//! Test all node costructors
+//! \brief \ref requirement
+TEST_CASE("continue_node constructors"){
+    using namespace oneapi::tbb::flow;
     graph g;
-    continue_node<int> fn(g, b);
 
-    CopyCounterBody<int> b2 = copy_body<CopyCounterBody<int>,
-                                             continue_node<int>>(fn);
+    conformance::counting_functor<int> fun;
+
+    continue_node<int> proto1(g, fun);
+    continue_node<int> proto2(g, fun, oneapi::tbb::flow::node_priority_t(1));
+    continue_node<int> proto3(g, 2, fun);
+    continue_node<int> proto4(g, 2, fun, oneapi::tbb::flow::node_priority_t(1));
 
-    CHECK_MESSAGE( (b.copy_count + 2 <= b2.copy_count), "copy_body and constructor should copy bodies");
+    continue_node<int, lightweight> lw_node1(g, fun, lightweight());
+    continue_node<int, lightweight> lw_node2(g, fun, lightweight(), oneapi::tbb::flow::node_priority_t(1));
+    continue_node<int, lightweight> lw_node3(g, 2, fun, lightweight());
+    continue_node<int, lightweight> lw_node4(g, 2, fun, lightweight(), oneapi::tbb::flow::node_priority_t(1));
 }
 
+//! The node that is constructed has a reference to the same graph object as src,
+//! has a copy of the initial body used by src, and only has a non-zero threshold if src is constructed with a non-zero threshold..
+//! The predecessors and successors of src are not copied.
+//! \brief \ref interface
+TEST_CASE("continue_node copy constructor"){
```

Comment:
This is the test have more special logic as corresponding test from `conformance` because of `number_of_predecessors` constructor parameter of continue_node.